### PR TITLE
Apache auto-instrumentation

### PR DESCRIPTION
**Pull request by chrlic**
PR proposes auto-instrumentation for applications running on Apache HTTPD server in K8S environment.
For instrumentation library, it uses https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/otel-webserver-module
For instrumentation itself, it adheres to the principles used for other languages already supported as much as possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -286,11 +291,31 @@ spec:
...
   dotnet:
     image: your-customized-auto-instrumentation-image:dotnet
+  apache:
+    image: your-customized-auto-instrumentation-image:apache
 ```
 
 The Dockerfiles for auto-instrumentation can be found in [autoinstrumentation directory](./autoinstrumentation).
 Follow the instructions in the Dockerfiles on how to build a custom container image.
 
+>**Note:** For `Apache` auto-instrumentation, by default, instrumentation assumes httpd version 2.4. If you need to use version 2.2, or you need to adjust agent parameters, adjust instrumentation per following example:
```
Comment (pavolloffay) [inline README.md:301]: This deserves a new section

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -1,2 +1,8 @@
 resources:
-- manager.yaml
\ No newline at end of file
+- manager.yaml
```
Comment (pavolloffay) [inline config/manager/kustomization.yaml:2]: this needs to be reverted

**Comment by pavolloffay** [comment]
I would propose to split the PR into:
1. auto-instrumentation image and document the approach
2. create injection functionality with e2e test
3. setting the default version of auto-instrumentation in the operator

File: README.md (hunk 1 above)
Comment (chrlic) [reply README.md:301]: Hello Pavol,

thank you for your review. Based on you comments, I'd do a new set of PR's
1) auto-instrumentation image build and documentation
2) changes related to Instrumentation CRD - definition, validation, etc.
3) Injection itself + e2e tests
4) default settings in the operator

Would that be OK?

Comment (pavolloffay) [reply README.md:301]: yes, that would be perfect ❤️

**Comment by jaronoff97** [comment]
no longer needed! #1236 did this

**Pull request closed without merging**